Firestore/core/test/unit/remote/bloom_filter_test.cc

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Firestore/core/src/remote/bloom_filter.h"
+#include <vector>
+#include "gtest/gtest.h"
+
+namespace firebase {
+namespace firestore {
+namespace remote {
+namespace {
+
+using util::Status;
+using util::StatusOr;
+
+TEST(BloomFilterTest, CanInstantiateEmptyBloomFilter) {
+  BloomFilter bloom_filter = BloomFilter(std::vector<uint8_t>{}, 0, 0);
+  EXPECT_EQ(bloom_filter.bit_count(), 0);
+}
+
+TEST(BloomFilterTest, CanInstantiateNonEmptyBloomFilter) {
+  {
+    BloomFilter bloom_filter = BloomFilter(std::vector<uint8_t>{1}, 0, 1);
+    EXPECT_EQ(bloom_filter.bit_count(), 8);
+  }
+  {
+    BloomFilter bloom_filter = BloomFilter(std::vector<uint8_t>{1}, 7, 1);
+    EXPECT_EQ(bloom_filter.bit_count(), 1);
+  }
+}
+
+/* Test Create factory method with valid and invalid inputs */
+TEST(BloomFilterTest, CreateShouldReturnBloomFilterOnValidInputs) {
+  StatusOr<BloomFilter> maybe_bloom_filter =
+      BloomFilter::Create(std::vector<uint8_t>{1}, 1, 1);
+  ASSERT_TRUE(maybe_bloom_filter.ok());
+  BloomFilter bloom_filter = maybe_bloom_filter.ValueOrDie();
+  EXPECT_EQ(bloom_filter.bit_count(), 7);
+}
+
+TEST(BloomFilterTest, CreateShouldBeAbleToCreatEmptyBloomFilter) {
+  StatusOr<BloomFilter> maybe_bloom_filter =
+      BloomFilter::Create(std::vector<uint8_t>{}, 0, 0);
+  ASSERT_TRUE(maybe_bloom_filter.ok());
+  BloomFilter bloom_filter = maybe_bloom_filter.ValueOrDie();
+  EXPECT_EQ(bloom_filter.bit_count(), 0);
+}
+
+TEST(BloomFilterTest, CreateShouldReturnNotOKStatusOnNegativePadding) {
+  {
+    StatusOr<BloomFilter> maybe_bloom_filter =
+        BloomFilter::Create(std::vector<uint8_t>{0}, -1, 1);
```

Comment:
I think this test intends to test an empty vector... if so, replace `{0}` with `{}` because `{0}` creates a vector with exactly one elements whose value is `0`, and this is the same as the test in the next block.

Also on line 82.

Reply:
I was trying to create a BoomFilter with one byte, which is 0. But it is clearly misleading. I will change it to {1}.